numba/np/polynomial.py

```diff
@@ -178,18 +232,130 @@ def numpy_polymul(c1, c2):
         msg = 'The argument "c2" must be array-like'
         raise errors.TypingError(msg)
 
-    if np.ndim(c1) > 1 or np.ndim(c2) > 1:
-        msg = 'Coefficient array is not 1-d'
-        raise errors.NumbaValueError(msg)
+    def impl(c1, c2):
+        arr1, arr2 = pu.as_series((c1, c2))
+        val = np.convolve(arr1, arr2)
+        return pu.trimseq(val)
+
+    return impl
+
+
+@overload(poly.polyval)
+def poly_polyval(x, c, tensor=True):
+    if not type_can_asarray(x):
+        msg = 'The argument "x" must be array-like'
+        raise errors.TypingError(msg)
+
+    if not type_can_asarray(c):
+        msg = 'The argument "c" must be array-like'
+        raise errors.TypingError(msg)
+
+    if not isinstance(tensor, (bool, types.Literal, types.Boolean)):
+        msg = 'The argument "tensor" must be boolean'
+        raise errors.TypingError(msg)
+
+    if not isinstance(tensor, (bool, types.Literal)):
+        SentryLiteralArgs(['tensor']).for_function(poly_polyval).bind(x, c,
+                                                                      tensor)
+
+    res_dtype = _poly_result_dtype(c, x)
+
+    # Simulate new_shape = (1,) * np.ndim(x) in the general case
+    # If x is a number, new_shape is not used
+    # If x is a tuple or a list, then it's 1d hence new_shape=(1,)
+    x_nd_array = not isinstance(x, types.Number)
+    new_shape = (1,)
+    if isinstance(x, types.Array):
+        # If x is a np.array, then take its dimension
+        new_shape = (1,) * np.ndim(x)
+
+    if isinstance(tensor, bool):
+        tensor_arg = tensor
+    else:
+        tensor_arg = tensor.literal_value
+
+    def impl(x, c, tensor=True):
+        arr = np.asarray(c).astype(res_dtype)
+        inputs = np.asarray(x).astype(res_dtype)
+        if x_nd_array and tensor_arg:
+            arr = arr.reshape(arr.shape + new_shape)
+
+        l = len(arr)
+        y = arr[l - 1] + inputs * 0
+
+        for i in range(l - 1, 0, -1):
+            y = arr[i - 1] + y * inputs
+
+        return y
+
+    return impl
 
-    result_dtype = _poly_result_dtype((c1, c2))
+
+@overload(poly.polyint)
+def poly_polyint(c, m=1):
+
+    if not type_can_asarray(c):
+        msg = 'The argument "c" must be array-like'
+        raise errors.TypingError(msg)
+
+    if not isinstance(m, (int, types.Integer)):
+        msg = 'The argument "m" must be an integer'
+        raise errors.TypingError(msg)
+    res_dtype = _poly_result_dtype(c)
+    is1D = ((np.ndim(c) == 1) or
+            (isinstance(c, types.List) and isinstance(c.dtype, types.Number)))
```

Comment:
In case of tuple/list, you also need to check its dtype and raise an exception if it is not a number.

```python
from numba import njit
import numpy as np

pa = np.polynomial.polynomial

@njit
def foo(a):
    return pa.polyint(a)

a = ['a', 'b', 'c']
print(foo(a))
```